apps/checkout/src/sections/UserAddresses/AddressForm.tsx

```diff
@@ -0,0 +1,138 @@
+import { Button } from "@/components/Button";
+import { TextInput } from "@/components/TextInput";
+import { CountryCode, useAddressValidationRulesQuery } from "@/graphql";
+import { useErrorMessages } from "@/hooks/useErrorMessages";
+import { MessageKey, useFormattedMessages } from "@/hooks/useFormattedMessages";
+import { useGetInputProps } from "@/hooks/useGetInputProps";
+import { AddressField } from "@/lib/globalTypes";
+import {
+  ensureArray,
+  getRequiredAddressFields,
+  getSortedAddressFields,
+  useValidationResolver,
+} from "@/lib/utils";
+import { useErrorsContext } from "@/providers/ErrorsProvider";
+import forEach from "lodash/forEach";
+import { useEffect } from "react";
+import {
+  DefaultValues,
+  Path,
+  SubmitHandler,
+  UnpackNestedValue,
+  useForm,
+} from "react-hook-form";
+import { object, string } from "yup";
+import { AddressFormData } from "./types";
+
+interface AddressFormProps<TFormData extends AddressFormData> {
+  countryCode: CountryCode;
+  defaultValues?: DefaultValues<TFormData>;
+  onCancel?: () => void;
+  onSave: SubmitHandler<TFormData>;
+}
+
+export const AddressForm = <TFormData extends AddressFormData>({
```

Comment:
```suggestion
export const AddressForm = <TFormData extends AddressFormData>({
```
Do we need a generic here? Can't we just use `AddressFormData`?

Reply:
It's either addressFormData or userAddressFormData where userAddressFormData has `id` in it, that is crucial since we later on pass that to onSubmit for update function. I mean we could make a hack around that, but I don't see a value in developing hacks to make the types a bit easier